File: Api/RecipeApi/Recipe.Common/Models/RecipeModel.cs
namespace Recipe.Common.Models;

public class RecipeModel
{
    public int Id { get; init; }
    public required string Name { get; init; }
    public required string Description { get; init; }
    public required List<IngredientModel> Ingredients { get; init; }
    public string? Base64Image { get; init; } 
}

File: App/lib/models/domain/recipe_model.dart
import 'package:app/models/domain/ingredient_model.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
part 'recipe_model.g.dart';

@JsonSerializable(explicitToJson: true)
class RecipeModel {
  RecipeModel();
  RecipeModel.fromValues(this.name, this.description, this.ingredients, this.base64Image);

  int id = -1;
  String name = '';
  String description = '';
  List<IngredientModel> ingredients = [];
  String? base64Image;

  factory RecipeModel.fromJson(Map<String, dynamic> json) => _$RecipeModelFromJson(json);
  Map<String, dynamic> toJson() => _$RecipeModelToJson(this);
}
